horovod/data/data_loader_base.py

```diff
@@ -75,34 +76,57 @@ def close_async_loader(self):
         """
         Close the async data loader.
         """
-        print("Closing the AsyncDataLoaderMixin.")
+        print(f"close_async_loader[{self.async_loader_queue_size}], Closing the AsyncDataLoaderMixin.")
         if self.async_loader_queue_size > 0 and self.started:
             self.finished_event.set()
+            c = 0
             while True:
                 try:
                     # Drain buffer
                     self.queue.get_nowait()
+                    if self.debug_data_loader:
+                        print(f"close_async_loader[{self.async_loader_queue_size}], discarded batch #{c} from Queue.")
+
+                    c += 1
+
+                    # Force break out if hanging. We assume hanging if already pop enough items from queue.
+                    if c > max(2 * self.async_loader_queue_size, 200):
+                        print(f"close_async_loader: ERROR!!! Force break out after {c} times get_nowait.")
+                        break
                 except Empty:
                     break
+            print(f"close_async_loader[{self.async_loader_queue_size}], joining...")
```

Comment:
make these prints all under `debug_data_loader=True`

Reply:
yes, will add this.